internal/controller/deployment.go

```diff
@@ -28,13 +28,62 @@ func (r *TritonInterfaceServerReconciler) deploymentForModelServing(tis *aitoolk
 					CreationTimestamp: metav1.Now(),
 				},
 				Spec: corev1.PodSpec{
-					Volumes: []corev1.Volume{{Name: "persistent-volume", VolumeSource: corev1.VolumeSource{PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{ClaimName: tis.Spec.PvcName}}}},
+					Volumes: func() []corev1.Volume {
```

Comment:
This definition of function here is problematic. please refrain from using closures and include all if conditions in the main function itself. There is cost to define functions and we can't do it for 2 if conditions